### ISSUE TITLE
ShapeDataset seed has no effect: default_rng() ignores np.random.seed()

### ISSUE THREAD

**Issue by kavishsethi5-dev** [datawhalechina/learn-world-model#12]
In `docs/en/projects/p01_vae_encoder.ipynb`, `ShapeDataset.__init__` calls `np.random.seed(seed)`, but `make_shape_image` builds its own generator with `np.random.default_rng()` and no argument. `default_rng()` seeds itself from OS entropy and does not read the legacy global state that `np.random.seed` sets, so the seed has no effect and the dataset is regenerated differently on every run.

**Reproduction**

```python
a = ShapeDataset(n_samples=50)
b = ShapeDataset(n_samples=50)
print(torch.equal(a.images, b.images))       # False, expected True
print(torch.equal(a.images[0], a.images[1])) # False, correct
```

First line should be True with a working seed. Second line is the control and should stay False.

**Why it matters here**

P01 saves `vae_encoder.pt` for P02 and P03 to load, so the checkpoint is trained on a different dataset each run. Reconstruction and KL values, the latent traversal, and which latent dimensions end up active all shift between runs, which makes results non-comparable between readers.

**Proposed fix**

Passing `seed` straight into `make_shape_image` does not work, since it is called `n_samples` times and would reseed identically each call, giving n_samples copies of the same image. Creating one generator and passing it in keeps the draws sequential:

```python
def make_shape_image(img_size=64, rng=None):
    if rng is None:
        rng = np.random.default_rng()
    ...

class ShapeDataset(Dataset):
    def __init__(self, n_samples=1000, img_size=64, seed=42):
        torch.manual_seed(seed)
        rng = np.random.default_rng(seed)
        self.images = torch.stack(
            [make_shape_image(img_size, rng) for _ in range(n_samples)]
        )
```

The `rng=None` default keeps `make_shape_image` usable standalone. The now-unused `np.random.seed(seed)` line can go, since leaving it in is what makes the seeding look handled.

The same cell appears in the ZH and KO notebooks, so the fix would need applying in all three.

**Before I open a PR**

CLAUDE.md says not to touch notebook code cells unless asked, so I would rather check first: would you like a PR for this across all three notebooks, or would you prefer to patch it yourselves?

**Comment by zhimin-z** [comment]
> In `docs/en/projects/p01_vae_encoder.ipynb`, `ShapeDataset.__init__` calls `np.random.seed(seed)`, but `make_shape_image` builds its own generator with `np.random.default_rng()` and no argument. `default_rng()` seeds itself from OS entropy and does not read the legacy global state that `np.random.seed` sets, so the seed has no effect and the dataset is regenerated differently on every run.
> 
> **Reproduction**
> 
> a = ShapeDataset(n_samples=50)
> b = ShapeDataset(n_samples=50)
> print(torch.equal(a.images, b.images))       # False, expected True
> print(torch.equal(a.images[0], a.images[1])) # False, correct
> First line should be True with a working seed. Second line is the control and should stay False.
> 
> **Why it matters here**
> 
> P01 saves `vae_encoder.pt` for P02 and P03 to load, so the checkpoint is trained on a different dataset each run. Reconstruction and KL values, the latent traversal, and which latent dimensions end up active all shift between runs, which makes results non-comparable between readers.
> 
> **Proposed fix**
> 
> Passing `seed` straight into `make_shape_image` does not work, since it is called `n_samples` times and would reseed identically each call, giving n_samples copies of the same image. Creating one generator and passing it in keeps the draws sequential:
> 
> def make_shape_image(img_size=64, rng=None):
>     if rng is None:
>         rng = np.random.default_rng()
>     ...
> 
> class ShapeDataset(Dataset):
>     def __init__(self, n_samples=1000, img_size=64, seed=42):
>         torch.manual_seed(seed)
>         rng = np.random.default_rng(seed)
>         self.images = torch.stack(
>             [make_shape_image(img_size, rng) for _ in range(n_samples)]
>         )
> The `rng=None` default keeps `make_shape_image` usable standalone. The now-unused `np.random.seed(seed)` line can go, since leaving it in is what makes the seeding look handled.
> 
> The same cell appears in the ZH and KO notebooks, so the fix would need applying in all three.
> 
> **Before I open a PR**
> 
> CLAUDE.md says not to touch notebook code cells unless asked, so I would rather check first: would you like a PR for this across all three notebooks, or would you prefer to patch it yourselves?

Thanks, this is very sharp notice. Would you like to raise a patch, it is welcome if you could do so :)

**Comment by kavishsethi5-dev** [comment]
Will do!
